### PR TITLE
A4A: Add Partner directory onboarding card.

### PR DESCRIPTION
**Pull request by jkguidaven**
This pull request  adds the Partner directory onboarding card to the Overview page.
Screenshot 2024-06-18 at 10 33 39 PM

Closes https://github.com/Automattic/automattic-for-agencies-dev/issues/549

## Proposed Changes

* Implement a reusable Sticky card for future usage.
* Implement the Partner directory onboarding card based on the Sticky card component. Use the `a4a-partner-directory-onboarding-card` preference as the flag to display the onboarding card.

## Testing Instructions

* Use the A4A live link and go to the `/overview` page
* Confirm that the onboarding card is visible. Click any of the buttons and confirm the Onboarding card will no longer display.
   * Dismiss button
   * The **Maybe Later** button
   * The **Let's do it** button
* To reset the flow, use the Calypso debug tool at the bottom right corner of the screen and remove `a4a-partner-directory-onboarding-card` preference.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Review by andrii-lysenko — APPROVED** [review]
Looks good to me 👍

**Review by keoshi — APPROVED** [review]
Looks great!

This is super extra, but is there a change we could add a unveil animation to draw attention to this elements after landing on A4A?

https://github.com/Automattic/wp-calypso/assets/390760/c64996cd-a88d-4f9f-9101-3166b3a87efd

In case it's helpful, here's a Codepen showing what it could look like: https://codepen.io/keoshi/pen/MWdQpxj

**Comment by jkguidaven** [comment]
@keoshi, thanks for the snippet. Helps a lot.

https://github.com/Automattic/wp-calypso/assets/56598660/85916843-2388-42ca-adc7-62c7eccccb23

**Comment by keoshi** [comment]
That looks stunning @jkguidaven — thanks so much for including this!

File: client/a8c-for-agencies/components/sticky-card/index.tsx [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+type Props = {
+	className?: string;
+	position?: 'top left' | 'top right' | 'bottom left' | 'bottom right';
+	children: ReactNode;
+	dismissable?: boolean;
+	title?: string;
+	onClose?: () => void;
+};
+
+export function StickyCard( {
```
Comment (cleacos) [inline client/a8c-for-agencies/components/sticky-card/index.tsx:17]: ❤️

File: client/a8c-for-agencies/sections/overview/partner-directory-onboarding-card/hooks/use-partner-directory-onboard-card.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+	const hasPreference = useSelector( ( state ) =>
+		getPreference( state, PARTNER_DIRECTORY_ONBOARDING_CARD_PREFERENCE_NAME )
+	);
+
+	const hideCard = useCallback( () => {
+		dispatch( savePreference( PARTNER_DIRECTORY_ONBOARDING_CARD_PREFERENCE_NAME, true ) );
+	}, [ dispatch ] );
+
+	return {
+		isActive: ! hasPreference && isSectionNameEnabled( 'a8c-for-agencies-partner-directory' ),
```
Comment (cleacos) [inline client/a8c-for-agencies/sections/overview/partner-directory-onboarding-card/hooks/use-partner-directory-onboard-card.ts:21]: 👌

**Review by cleacos — APPROVED** [review]
LGTM!

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/14944796

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @jkguidaven for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/14944796).

**Pull request merged**